Do-It/Controller/DoItTableViewController.swift

```diff
@@ -0,0 +1,116 @@
+
+    var doits = [DoIt]()
+    @IBOutlet var tableView: UITableView!
+
+    override func viewWillAppear(_ animated: Bool) {
+
+        doits.append(DoIt(identifier: DoItId(),
+                          course: Course(name: "fake"),
+                          dueDate: Calendar.current.date(byAdding: .day, value: 30, to: Date())!,
+                          description: "description here",
+                          name: "YO",
+                          priority: DoItPriority.low,
+                          kind: DoItKind.homework))
+
+        doits.append(DoIt(identifier: DoItId(),
+                          course: Course(name: "fake"),
+                          dueDate: Calendar.current.date(byAdding: .hour, value: 20, to: Date())!,
+                          description: "description here",
+                          name: "YO",
+                          priority: DoItPriority.low,
+                          kind: DoItKind.homework))
+
+        doits.append(DoIt(identifier: DoItId(),
+                          course: Course(name: "fake"),
+                          dueDate: Calendar.current.date(byAdding: .minute, value: 10, to: Date())!,
+                          description: "description here",
+                          name: "DIFF",
+                          priority: DoItPriority.low,
+                          kind: DoItKind.homework))
+
+        tableView.delegate = self
+        tableView.dataSource = self
+
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+
+    // MARK: - Table view data source
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return doits.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "DoItTableViewCell",
+                                                       for: indexPath) as? DoItTableViewCell else {
+            return tableView.dequeueReusableCell(withIdentifier: "DoItTableViewCell", for: indexPath)
+        }
+
+        cell.titleLabel?.text = doits[indexPath.item].name
+        cell.courseLabel?.text = doits[indexPath.item].course.name
+        cell.dateLabel?.text = formatted_date(index: indexPath.item)
+        cell.descriptionLabel?.text = doits[indexPath.item].description
+
+        return cell
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return CGFloat(80)
+    }
+
+    func formatted_date(index: Int) -> String? {
+        let diffDateComponents = Calendar.current.dateComponents([.day, .hour, .minute], from: Date(),
+                                                                 to: doits[index].dueDate)
+
+        let formatter = DateComponentsFormatter()
```

Comment:
Formatters can be expensive to create. Save the formatter as instance variable rather than initializing a new one each time we format a date; you can just reconfigure `allowUnits` each time it's used.